### ISSUE TITLE
spaCy Morphimizers Not Shown

### ISSUE THREAD

**Issue by drkthomp** [mortii/anki-morphs#234]
### Describe the bug

[//]: # (Please provide a clear and concise description of what the bug is.)

I installed spaCy morphimizers [as per the instructions in the guide](https://mortii.github.io/anki-morphs/user_guide/installation/installing-spacy.html) but they do not show up in Anki after restarting. Notable is that the Mecab morphimizer does. 

### Steps to reproduce the behavior
```
cd ~/.local/share/Anki2/addons21/
python3.12 -m venv spacyenv
source spacyenv/bin/activate
python -m pip install --upgrade pip setuptools wheel
python -m pip install --upgrade spacy six
python -m spacy download ja_core_news_lg
python -m spacy download da_core_news_lg
python -m spacy download es_core_news_md
deactivate
```

Restarting Anki and looked at morphimizers, but it is not shown. 

### Expected behavior

[//]: # (Please provide a clear and concise description of what would expected to happen instead.)
The spaCy morphimizers should be shown in ankimorphs upon restart. 
### Screenshots

[//]: # (If applicable, add screenshots to help explain your problem, otherwise you can delete this section.)
![2024-05-18T10:27:22,946961210+02:00](https://github.com/mortii/anki-morphs/assets/4887015/0860d600-9c55-4fe8-9047-dba14b36b15a)

### My setup

[//]: # (Note: the AnkiMorphs version is found on the bottom of the AnkiMorphs settings window)

[//]: # (Example:)
[//]: # (- Operating System: Windows 11)
[//]: # (- Anki Version: 23.12.1)
[//]: # (- AnkiMorphs Version: v2.0.0)

 - Operating System: Arch Linux 
 - Anki Version: Version ⁨24.04.1 (ccd9ca1a)⁩
Python 3.12.3 Qt 6.7.0 PyQt 6.7.0
 - AnkiMorphs Version: 2.2.4 

### Additional Info 
spAcy info:
```
============================== Info about spaCy ==============================

spaCy version    3.7.4                         
Location         /home/dranion/.local/share/Anki2/addons21/spacyenv/lib/python3.12/site-packages/spacy
Platform         Linux-6.8.9-zen1-2-zen-x86_64-with-glibc2.39
Python version   3.12.3                        
Pipelines        da_core_news_lg (3.7.0), ja_core_news_lg (3.7.0), xx_sent_ud_sm (3.7.0), es_core_news_md (3.7.0)
```

**Comment by mortii** [comment]
Weird, it looks correct...

Could you add some debugging print statements here:
https://github.com/mortii/anki-morphs/blob/f04cd88a7122f57460ec4e1190c7d3ad009ed2ca/ankimorphs/spacy_wrapper.py#L334 

e.g.:
```python
spacy_path = os.path.join(mw.pm.addonFolder(), "spacyenv")
print(f"spacy_path: {spacy_path}")
```
When you run anki from the terminal and try to change morphemizer you should see the output.

**Comment by drkthomp** [comment]
`spacy_path: /home/dranion/.local/share/Anki2/addons21/spacyenv`

**Comment by drkthomp** [comment]
It is hitting an exception at some point, probably in the if statement? Will check closer. 
```
spacy_path: /home/dranion/.local/share/Anki2/addons21/spacyenv
win is not true
Exception module not found
```

```
          spacy_path = os.path.join(mw.pm.addonFolder(), "spacyenv")
            print(f"spacy_path: {spacy_path}")

            if is_win is True:
                print("win is true")
                spacy_bin_path = os.path.join(spacy_path, "Scripts")
                spacy_site_packages_path = os.path.join(
                    spacy_path, "Lib", "site-packages"
                )
            else:
                print("win is not true")
                spacy_bin_path = os.path.join(spacy_path, "bin")
                spacy_site_packages_path = os.path.join(
                    spacy_path, "lib", "python3.9", "site-packages"
                )

            sys.path.append(spacy_bin_path)
            sys.path.append(spacy_site_packages_path)
            updated_python_path = True

        import spacy.util  # pylint:disable=import-outside-toplevel
        print(f"models: {spacy.util.get_installed_models()}")
        print([f"{model_name}" for model_name in spacy.util.get_installed_models()])
        return [f"{model_name}" for model_name in spacy.util.get_installed_models()]
    except ModuleNotFoundError:
        print("Exception module not found")
        # spacy not installed
        return []
```

**Comment by Tartee** [comment]
I had the same issue and was able to fix it. I am using `anki-bin 24.04.1-1` (with python 3.12) and had to change 

https://github.com/mortii/anki-morphs/blob/f04cd88a7122f57460ec4e1190c7d3ad009ed2ca/ankimorphs/spacy_wrapper.py#L344

to use `python3.12` instead of `python3.9`. Maybe there's a way to fetch anki's version of python and use that directly instead of hardcoding a version?

**Comment by drkthomp** [comment]
Yes, I just attempted that and it also fixed it for me. Seems like it is the hardcoding.

**Comment by mortii** [comment]
@Tartee My goodness, you're a hero! I'll patch it and push a hotfix asap, thanks!

**Comment by mortii** [comment]
Released in [2.2.5](https://github.com/mortii/anki-morphs/releases/tag/v2.2.5). Let me know if it didn't work or is bugged in any way :)
